### DISCUSSION TITLE
Treat any *.11tydata.js file as a directory data file. (👍10)

### DISCUSSION THREAD

**Discussion by robdodson** [11ty/eleventy#4027] [Enhancement Queue]
There have been a few instances where I wished that directory data files did not have to be named the same thing as the directory itself.

For example, I wanted to put a table of contents file inside of each of my category folders:

```
/performance
  toc.11tydata.js
  post-one/
  post-two/
/security
  toc.11tydata.js
  post-one/
  post-two/
```

Also, because we localize our site, our directory structure looks like this:

```
en/
es/
ja/
```

So the directory data file that I'm using for my homepage has to be called `en.11tydata.js` but I'd much rather just call it `home.11tydata.js` so it doesn't seem like it's related to the language itself.

**Comment by reubenlillie** [comment]
It'd be great if this could apply to JSON files too. For example, allow for `DIRECTORY/index.json` instead of insisting upon `DIRECTORY/directory.json`.

**Comment by reubenlillie** [comment]
⬆️ Seems more framework agnostic to me.

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by zachleat** [comment]
Some overlap here with the new `dataFileDirBaseNameOverride` feature #1699 that allows you to override the default base name file name we use for directory data file lookups.

**Comment by zachleat** [comment]
Related to #3919

**Reply by zachleat** [reply]
Marginally related to #4077

**Comment by darthmall** [comment]
This is an old one, so I don’t know if this is actually under consideration or not (it came on my radar because of the housekeeping Zach’s been doing), but this feels a bit risky to me. Consider the following situation:

```
/performance
  post-one.11tydata.js
  post-one.md
  post-two.md
```

Today, `post-one.11tydata.js` is a data file for the `post-one.md` template. With this change, it would suddenly become a directory data file and all of its contents would cascade into `post-two.md` as well. This seems undesirable.

Even if it’s not undesirable, this is a change that seems likely to wreak havoc on a lot of existing projects.

**Comment by cristovaov** [comment]
just happen to pass by, so idk what the latest is on this topic but i'd agree with @darthmall the current mechanism works fine imo. the break would be very much undesirable at this point in time.
